imputation/ts_impute.go

```diff
@@ -47,61 +57,130 @@ func dateGapFinder(keys []string) (string, int, int, int) {
 	case 10:
 		parseFormat = dayfmt
 	}
-
 	year, month, day := 0, 0, 0
 
-	duration := 1<<31 - 1 // A large number.
+	if parseFormat == "" {
+		return parseFormat, year, month, day, false, errors.New("The date format is not ISO 8601.")
+	}
```

Comment:
Please make this the default clause:

```suggestion
default:
  return parseFormat, year, month, day, false, errors.New("The date format is not ISO 8601.")
```